tests/unit_tests/data_model/test_metadata_model.py

```diff
@@ -29,6 +29,19 @@ def test_load_schema():
     with pytest.raises(FileNotFoundError):
         metadata_model._load_schema(schema_file="not_existing_file")
 
+    schema_file = files("simtools") / "schemas" / "model_parameter.metaschema.yml"
+    # schema versions
+    with pytest.raises(ValueError, match=r"^Schema version not given in"):
+        metadata_model._load_schema(schema_file)
+
+    _schema_1, _ = metadata_model._load_schema(schema_file, "0.1.0")
+    assert _schema_1["version"] == "0.1.0"
+    _schema_2, _ = metadata_model._load_schema(schema_file, "0.2.0")
+    assert _schema_2["version"] == "0.2.0"
+
+    with pytest.raises(ValueError, match=r"^Schema version 0.2 not found in"):
+        metadata_model._load_schema(schema_file, "0.2")
+
```

Comment:
Would be good to also check for the warning from `_logger.warning(f"Schema version {schema_version} does not match {schema.get('version')}")`

Reply:
Good point, added a test.